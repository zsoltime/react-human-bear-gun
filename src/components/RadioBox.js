import React from 'react';
import PropTypes from 'prop-types';

import Icon from './Icons';
import { stringOrNull } from '../helpers';

const RadioBox = ({ choice, userChoice, onChange }) => (
  <>
    <input
      id={choice}
      checked={choice === userChoice}
      className="radio visuallyhidden"
      onChange={onChange}
      type="radio"
    />
    <label
      className={
        choice === userChoice
          ? 'radio-box radio-box--active'
          : 'radio-box'
      }
      htmlFor={choice}
    >
      <Icon
        aria-hidden="true"
        className="icon"
        focusable="false"
        type={choice}
      />
      <span className="visuallyhidden">{choice}</span>
    </label>
  </>
);

RadioBox.propTypes = {
  choice: stringOrNull,
  userChoice: stringOrNull,
  onChange: PropTypes.func.isRequired,
};

export default RadioBox;
