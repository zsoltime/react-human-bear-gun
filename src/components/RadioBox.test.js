import React from 'react';
import { shallow } from 'enzyme';

import RadioBox from './RadioBox';

const CHOICES = ['Human', 'Bear', 'Gun'];

describe('<RadioBox />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <RadioBox
        choice={CHOICES[0]}
        userChoice={CHOICES[1]}
        onChange={console.log}
      />
    );

    expect(wrapper.find('Fragment')).toMatchSnapshot();
  });

  it('renders the correct icon', () => {
    const wrapper = shallow(
      <RadioBox
        choice={CHOICES[0]}
        userChoice={CHOICES[1]}
        onChange={console.log}
      />
    );

    expect(wrapper.find('Icon').prop('type')).toBe(CHOICES[0]);
  });

  it('renders the checkbox as checked if choice is the same as userChoice', () => {
    const wrapper = shallow(
      <RadioBox
        choice={CHOICES[1]}
        userChoice={CHOICES[1]}
        onChange={console.log}
      />
    );

    expect(wrapper.find('input').prop('checked')).toBe(true);
    expect(wrapper.find('label').prop('className')).toMatch('active');
  });

  it('calls onChange prop on change', () => {
    const handleChange = jest.fn();

    const wrapper = shallow(
      <RadioBox
        choice={CHOICES[0]}
        userChoice={CHOICES[2]}
        onChange={handleChange}
      />
    );

    wrapper.find('input').simulate('change');
    expect(handleChange).toBeCalledTimes(1);
  });
});
