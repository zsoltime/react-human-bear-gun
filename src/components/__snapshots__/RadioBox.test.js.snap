// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioBox /> matches snapshot 1`] = `
<Fragment>
  <input
    checked={false}
    className="radio visuallyhidden"
    id="Human"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="radio-box"
    htmlFor="Human"
  >
    <Icon
      aria-hidden="true"
      className="icon"
      focusable="false"
      type="Human"
    />
    <span
      className="visuallyhidden"
    >
      Human
    </span>
  </label>
</Fragment>
`;
